core/src/main/scala/org/apache/spark/storage/TachyonBlockManager.scala

```diff
@@ -95,8 +94,23 @@ private[spark] class TachyonBlockManager() extends ExternalBlockManager with Log
   override def putBytes(blockId: BlockId, bytes: ByteBuffer): Unit = {
     val file = getFile(blockId)
     val os = file.getOutStream(WriteType.TRY_CACHE)
-    os.write(bytes.array())
-    os.close()
+    try {
+      os.write(bytes.array())
+      os.close()
```

Comment:
close() does not have to be in a finally block, because cancel() will also close the block file.
This will cancel the block when a runtime exception happens.
I will add log for it. Thanks!